scripts/ci/Makefile

```diff
@@ -45,8 +46,19 @@ else
 	CONTAINER_OPTS := --rm -v /lib/modules:/lib/modules --tmpfs /run
 endif
 
+ifeq ($(CONTAINER_RUNTIME),podman)
+	# Just as Docker needs to use devicemapper Podman needs vfs
+	# as graphdriver as overlayfs does not support all test cases
+	STORAGE_DRIVER := vfs
+	# Podman limits the number of processes in a container using cgroups.
+	# Disable it as it breaks the thread-bomb test
+	CONTAINER_OPTS += --pids-limit=0
+endif
+
+export STORAGE_DRIVER
+
 restart-docker:
-	if [ "$$UNAME" = "x86_64" ]; then \
+	if [ "$$UNAME" = "x86_64" ] && [ -d /etc/docker ]; then \
```

Comment:
Not sure about this hunk... Doesn't it check that docker was not yet started? And we put it into "restart-docker:" target which is a prerequisite for "$(TARGETS):" target and "fedora-asan:" so it would probably be called _before_ we call "$(CONTAINER_RUNTIME)" recipe and would never restart docker and never add device-mapper graph driver...

Sorry I'm not too good at makefiles and probably missing something.

Reply:
It is a bit ugly. You are right. But now I am also not sure about it :wink: 

The goal of restart docker is to reconfigure it to use devicemapper instead of overlayfs for our tests. If I am running on a system without docker installed (i.e. /etc/docker does not exist), we are skipping the reconfiguration of the the docker daemon.

In the case of fedora-rawhide which should now be using Podman, because docker fails with the latest glibc in Fedora rawhide because of the blocked clone3() by seccomp, we are no longer using docker, so that docker-restart should be a noop if /etc/docker does not exist. This assumes if /etc/docker does not exist, restarting of the docker daemon will fail.

So in this case I am running the test with CONTAINER_RUNTIME=podman and if /etc/docker does not exist it should not fail restarting docker and thus failing the complete test.

I think this does the right thing, but maybe it is a bit ugly. It would not be really necessary for the CI environment where we have docker and podman installed, so even if we run with CONTAINER_RUNTIME=podman we can still restart the docker daemon even if it is totally unnecessary. I added this change because on my system, without docker installed, it failed restarting the docker daemon and that's why I added the check for /etc/docker.

Maybe a cleaner solution would be to check if CONTAINER_RUNTIME is == docker and only then reconfigure the docker daemon.

Reply:
I updated the PR to check if CONTAINER_RUNTIME is docker and only then reconfigure it.